homeassistant/components/shelly/light.py

```diff
@@ -55,10 +75,20 @@ def __init__(self, wrapper: ShellyDeviceWrapper, block: Block) -> None:
         super().__init__(wrapper, block)
         self.control_result = None
         self._supported_features = 0
-        if hasattr(block, "brightness"):
+        self._color_mode = (
+            hasattr(block, "red")
+            and hasattr(block, "green")
+            and hasattr(block, "blue")
+            and wrapper.device.settings["mode"] == "color"
+        )
+        if hasattr(block, "brightness") or hasattr(block, "gain"):
             self._supported_features |= SUPPORT_BRIGHTNESS
-        if hasattr(block, "colorTemp"):
+        if hasattr(block, "colorTemp") and not self._color_mode:
```

Comment:
Why would we leave color temp support out if there is color mode ? 

In a bunch of lights that's different leds.

Reply:
`colorTemp ` as no effect for Shelly if device is in color mode. Works fine instead on white mode

Reply:
Supported features need to reflect all possible supported features, not the ones currently supported based on the mode.